concrete/src/booleans/parameters.rs

```diff
@@ -1,135 +1,80 @@
-use concrete_boolean::parameters::BooleanParameters;
+use concrete_boolean::parameters::{
+    BooleanParameters, DecompositionBaseLog, DecompositionLevelCount, GlweDimension, LweDimension,
+    PolynomialSize, StandardDev,
+};
+pub use concrete_boolean::parameters::{DEFAULT_PARAMETERS, TFHE_LIB_PARAMETERS};
 
-pub trait BooleanParameterSet: Into<BooleanParameters> {}
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
 
-impl BooleanParameterSet for BooleanParameters {}
-
-#[cfg(feature = "__newer_booleans")]
-pub use new::FheBoolParameters;
-#[cfg(not(feature = "__newer_booleans"))]
-pub use old::FheBoolParameters;
-
-#[cfg(not(feature = "__newer_booleans"))]
-mod old {
-    use concrete_boolean::parameters::BooleanParameters;
-    pub use concrete_boolean::parameters::{DEFAULT_PARAMETERS, TFHE_LIB_PARAMETERS};
-    #[cfg(feature = "serde")]
-    use serde::{Deserialize, Serialize};
-
-    #[cfg_attr(doc, cfg(feature = "booleans"))]
-    #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-    #[derive(Clone, Debug, PartialEq)]
-    pub struct FheBoolParameters(pub(crate) BooleanParameters);
-
-    impl FheBoolParameters {
-        pub fn concrete_lib() -> Self {
-            Self::from_static(&DEFAULT_PARAMETERS)
-        }
-
-        pub fn tfhe_lib() -> Self {
-            Self::from_static(&TFHE_LIB_PARAMETERS)
-        }
+pub trait BooleanParameterSet: Into<BooleanParameters> {
+    type Id: Copy;
+}
 
-        fn from_static(params: &'static BooleanParameters) -> Self {
-            Self(*params)
-        }
-    }
+/// Parameters for [FheBool].
+#[cfg_attr(doc, cfg(feature = "booleans"))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct FheBoolParameters {
+    pub lwe_dimension: LweDimension,
+    pub glwe_dimension: GlweDimension,
+    pub polynomial_size: PolynomialSize,
+    pub lwe_modular_std_dev: StandardDev,
+    pub glwe_modular_std_dev: StandardDev,
+    pub pbs_base_log: DecompositionBaseLog,
+    pub pbs_level: DecompositionLevelCount,
+    pub ks_base_log: DecompositionBaseLog,
+    pub ks_level: DecompositionLevelCount,
+}
 
-    impl From<FheBoolParameters> for BooleanParameters {
-        fn from(params: FheBoolParameters) -> Self {
-            params.0
-        }
+impl FheBoolParameters {
+    pub fn concrete_lib() -> Self {
+        Self::from_static(&DEFAULT_PARAMETERS)
     }
 
-    impl From<BooleanParameters> for FheBoolParameters {
-        fn from(params: BooleanParameters) -> FheBoolParameters {
-            Self(params)
-        }
+    pub fn tfhe_lib() -> Self {
+        Self::from_static(&TFHE_LIB_PARAMETERS)
     }
 
-    impl Default for FheBoolParameters {
-        fn default() -> Self {
-            Self::concrete_lib()
-        }
+    fn from_static(params: &'static BooleanParameters) -> Self {
```

Comment:
I made this require a static lifetime because this function is meant to be called only on `const` defined
in concrete-boolean such as
```rust
pub const DEFAULT_PARATEMETERS: BooleanParameters {
  // ...
}
```

But other than that there is no reason